catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PointToBrickAdditionalTest.java

```diff
@@ -0,0 +1,190 @@
+import static org.catrobat.catroid.uiespresso.util.matchers.BundleMatchers.bundleHasExtraIntent;
+import static org.catrobat.catroid.uiespresso.util.matchers.BundleMatchers.bundleHasMatchingString;
+import static org.hamcrest.core.AllOf.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class PointToBrickAdditionalTest {
+	private Matcher<Intent> expectedChooserIntent;
+	private Matcher<Intent> expectedGetContentIntent;
+	private final String lookFileName = "catroid_sunglasses.png";
+	private final String projectName = getClass().getSimpleName();
+	private final File tmpPath = new File(
+			Environment.getExternalStorageDirectory().getAbsolutePath(), "Pocket Code Test Temp");
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+	@Rule
+	public GrantPermissionRule runtimePermissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(this.getClass().getSimpleName());
+		baseActivityTestRule.launchActivity();
+		Intents.init();
+
+		expectedGetContentIntent = allOf(
+				hasAction("android.intent.action.GET_CONTENT"),
+				hasType("image/*"));
+
+		String chooserTitle = UiTestUtils.getResourcesString(R.string.select_look_from_gallery);
+		expectedChooserIntent = allOf(
+				hasAction("android.intent.action.CHOOSER"),
+				hasExtras(bundleHasMatchingString("android.intent.extra.TITLE", chooserTitle)),
+				hasExtras(bundleHasExtraIntent(expectedGetContentIntent)));
+
+		if (!tmpPath.exists()) {
+			tmpPath.mkdirs();
+		}
+
+		File imageFile = ResourceImporter.createImageFileFromResourcesInDirectory(
+				InstrumentationRegistry.getContext().getResources(),
+				org.catrobat.catroid.test.R.drawable.catroid_banzai,
+				tmpPath,
+				lookFileName,
+				1);
+
+		Intent resultData = new Intent();
+		resultData.setData(Uri.fromFile(imageFile));
+
+		Instrumentation.ActivityResult result =
+				new Instrumentation.ActivityResult(Activity.RESULT_OK, resultData);
+
+		intending(expectedChooserIntent).respondWith(result);
+	}
+
+	@Test
+	public void testAllPointTowardsAndCloneBricksSpinnersShowTheNewAddedObject() {
+		onBrickAtPosition(1)
+				.onSpinner(R.id.brick_point_to_spinner)
+				.performSelectNameable(R.string.new_option);
+
+		onView(withId(R.id.dialog_new_look_gallery))
+				.perform(click());
+
+		intended(expectedChooserIntent);
+
+		onView(withText(R.string.new_sprite_dialog_title))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.ok))
+				.perform(click());
+
+		String newObjectName = lookFileName.replace(".png", "");
+		checkBricksSpinnerHasAndShowsText(R.id.brick_point_to_spinner, 2, newObjectName);
+		checkBricksSpinnerHasAndShowsText(R.id.brick_clone_spinner, 3, newObjectName);
```

Comment:
the clone brick should be in its own separate test.

Reply:
I disagree, since I'm testing if the values of the spinner get updated after adding a new look using another brick "spinner of clone brick has no 'New..' option"